compiler/src/iree/compiler/Dialect/LinalgExt/IR/AggregatedOpInterfaceImpl.cpp

```diff
@@ -287,6 +287,9 @@ OnlineAttentionOp::decomposeOperation(OpBuilder &b) {
   Value oldSum = getSum();
   Type elementType = getElementTypeOrSelf(getOutput().getType());
 
+  auto qkAttrs = (*this)->getAttrOfType<DictionaryAttr>("qk_attrs");
+  auto pvAttrs = (*this)->getAttrOfType<DictionaryAttr>("pv_attrs");
```

Comment:
It is better to declare constant variables for `qk/pv_attrs` because they are also used in `KernelConfig.cpp`. Also, we can get rid of magic numbers/strings. We can declare the constant variables in [MarkerUtils.h](https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/Utils/MarkerUtils.h). What do you think?

Reply:
Sounds good. I did not want them to be part of the operation definition because they are codegen specific markers. MarkerUtils sounds like a good place for them.

Reply:
I don't think we can do that. MarkerUtils is part of Codegen/, while LinalgExt is part of Dialects/. I can make it part of the attention op definition as well I guess.

Reply:
Good point, I missed that! Can we add some getters to `extraClassDeclaration`? The concern is that the magic strings are floating in the codebase, which looks brittle to me. It looks better if we can centralize to somewhere, e.g., we can do something like:

```cpp

// They share the same entry which could be "config".
void setConfig(DictionaryAttr attr) { ... }
DictionaryAttr getConfig() { return (*this)->getAttrOfType<DictionaryAttr>("config"); }
bool hasConfig(StringRef str) {
  auto config = getConfig();
  if (!config) return false;
  return config.getNamed(str);
}

// Marker-like things
static StringRef getQKAttrStr() { return "qk_attrs"; }
static StringRef getPVAttrStr() { return "pv_attrs"; }

```

https://github.com/iree-org/iree/blob/0c2c627747586ed39ce7b1f6bfc9d8b83c4a4e69/compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.td#L530-L561

Reply:
I added a decompositon_config attribute to AttentionOp, which acts as the config attribute you asked for above. I also added getQKAttrStr/getPVAttrStr to extraClassDeclarations.